panels/exo_skel.py

```diff
@@ -107,6 +107,27 @@ def execute(self, context):
         bpy.ops.object.mode_set(mode=current_mode)
         return {'FINISHED'}
 
+class PopulateBoneList(bpy.types.Operator):
+    bl_idname = 'sub.populate_bone_list'
+    bl_label = 'Auto Populate Bone List'    
+    bl_description = 'Automatically assign a smash bone to an entry if its name matches with the prefix removed.'
+    bl_options = {'REGISTER', 'UNDO', 'INTERNAL'}
+    
+    def execute(self, context):
+        current_mode = context.object.mode
+        bpy.ops.object.mode_set(mode='OBJECT')
+
+        armature_smash = get_smash_armature()
+        prefix = bpy.context.scene.armature_prefix
+        
+        for bone_item in context.scene.bone_list:
+            #Auto populate list if bone's name without the prefix is in the Smash Armature.
+            if bone_item.bone_name_other.replace(prefix,'',1) in armature_smash.data.bones and not bone_item.bone_name_smash:
```

Comment:
I think you can use `.replaceprefix(prefix)` here instead. We only want to replace the prefix if it occurs at the start of the string. If that isn't available, you should be able to do it by checking if the string starts with the prefix and removing the prefix using string slicing.